src/generator/models.ts

```diff
@@ -244,7 +244,32 @@ function generateStructs(objects?: ObjectSchema[]): StructDef[] {
       text += `\tmsg := ""\n`;
       for (const prop of values(structDef.Properties)) {
         text += `\tif e.${prop.language.go!.name} != nil {\n`;
-        text += `\t\tmsg += fmt.Sprintf("${prop.language.go!.name}: %v\\n", *e.${prop.language.go!.name})\n`;
+        // check if the property is an object or a basic type and output the corresponding error message.
+        // this will only include details for the top level information in the corresponding type. 
+        if (!isObjectType(prop.schema)) {
+          text += `\t\tmsg += fmt.Sprintf("${prop.language.go!.name}: %v\\n", *e.${prop.language.go!.name})\n`;
+        } else {
+          // if the property is an object schema add the information in the Go struct to the error message 
+          if (prop.language.go!.errorType) {
```

Comment:
You could condense this a bit by moving this `if` block up as an `else if` on line 251.

Reply:
Also curious if we have an example of this in any swaggers?

Reply:
I havent seen this in any of the swaggers. I added it in case it does occur. My thought was that these nicer error messages would be embedded into each other if it ever happened. But I could leave it unimplemented OR throw an error if it ever does happen so we can take a look at it then?

Reply:
No you can leave it, will be handy if that ever happens.

Reply:
Done, fixed the if statement.